src/Interpreters/DDLWorker.cpp

```diff
@@ -1046,6 +1051,10 @@ String DDLWorker::enqueueQuery(DDLLogEntry & entry)
     zookeeper->createAncestors(query_path_prefix);
 
     String node_path = zookeeper->create(query_path_prefix, entry.toString(), zkutil::CreateMode::PersistentSequential);
+    DB::ReadBufferFromString in(node_path.substr(query_path_prefix.length()));
+    UInt64 id;
+    readText(id, in);
+    CurrentMetrics::set(*max_pushed_entry_metric, id);
```

Comment:
Shouldn't we set `max(*max_pushed_entry_metric, id)`?

Reply:
@Avogar 
yeah, updated.

Reply:
And I guess we should do it under `if (max_pushed_entry_metric)`

Reply:
sure, updated.

Reply:
@Avogar 
sure, udpated.

Reply:
We still have the problem with nullptr in `CurrentMetrics::set(*max_pushed_entry_metric, id);`. Let's just move all logic from lines 1054-1059 under `if (max_pushed_entry_metric)`

Reply:
@Avogar 
sorry for the back-and-forth change, just updated, please check again.